src/main/java/gov/cms/madie/cqllibraryservice/services/CqlLibraryService.java

```diff
@@ -111,8 +115,59 @@ public Page<LibraryListDTO> getLibrariesByCriteria(
       OwnershipType ownershipType,
       Pageable pageReq,
       String username) {
-    return cqlLibraryRepository.searchLibrariesByCriteria(
-        username, pageReq, librarySearchCriteria, ownershipType);
+
+    Page<LibraryListDTO> librariesPage =
+        cqlLibraryRepository.searchLibrariesByCriteria(
+            username, pageReq, librarySearchCriteria, ownershipType);
+    if (appConfigService.isFlagEnabled(MadieFeatureFlag.DISPLAY_OWNER)) {
+      log.debug("Enriching {} libraries with user details", librariesPage.getContent().size());
+      enrichWithUserDetails(librariesPage.getContent());
+    }
+    return librariesPage;
+  }
+
+  private void enrichWithUserDetails(List<LibraryListDTO> libraries) {
+    if (CollectionUtils.isEmpty(libraries)) {
+      log.debug("No libraries to enrich");
+      return;
+    }
+
+    // Extract unique owner HARP IDs
+    List<String> ownerIds =
+        libraries.stream()
+            .map(lib -> lib.getLibrarySet() != null ? lib.getLibrarySet().getOwner() : null)
+            .filter(Objects::nonNull)
+            .distinct()
+            .collect(Collectors.toList());
+
+    log.debug("Found {} unique owner IDs: {}", ownerIds.size(), ownerIds);
+
+    if (ownerIds.isEmpty()) {
+      log.debug("No owner IDs found to fetch user details");
+      return;
+    }
+
+    // Fetch user details in bulk
+    Map<String, UserDetailsDto> userDetailsMap = userServiceClient.getBulkUserDetails(ownerIds);
+
+    // Enrich each measure with user details
+    libraries.forEach(
+        library -> {
+          if (library.getLibrarySet() != null && library.getLibrarySet().getOwner() != null) {
+            String ownerId = library.getLibrarySet().getOwner();
+            UserDetailsDto userDetails = userDetailsMap.get(ownerId);
+
+            if (userDetails != null) {
+              library.setOwner(getFullName(userDetails));
+            }
+          }
+        });
+  }
+
+  private String getFullName(UserDetailsDto userDetails) {
+    String firstName = userDetails.getFirstName() != null ? userDetails.getFirstName() : "";
+    String lastName = userDetails.getLastName() != null ? userDetails.getLastName() : "";
+    return firstName + " " + lastName;
```

Comment:
in rare case it append unnecessary empty string to the full name. How about this?
```suggestion
    String firstName = StringUtils.defaultString(userDetails.getFirstName());
    String lastName = StringUtils.defaultString(userDetails.getLastName());
    return Stream.of(firstName, lastName)
        .filter(StringUtils::isNotBlank)
        .collect(Collectors.joining(" "))
        .trim();
```
OR

```suggestion
    String firstName = userDetails.getFirstName() != null ? userDetails.getFirstName() : "";
    String lastName = userDetails.getLastName() != null ? userDetails.getLastName() : "";
    return (firstName + " " + lastName).trim();
```

Reply:
I was doing the trim on the frontend side that is the reason i wasn't doing it on the backend side